src/ModelFactory.php

```diff
@@ -95,6 +92,9 @@ final public static function findOrCreate(array $attributes): Proxy
     /**
      * @see RepositoryProxy::first()
      *
+     * @return Proxy<TModel>&TModel
+     * @psalm-return Proxy<TModel>
```

Comment:
Psalm will (correctly) complain on `@return Proxy<TModel>&TModel`. That is why we need tell psalm (only psalm) the return type is actually `Proxy<TModel>`.

Reply:
Same as previous, is there a need for `&TModel`? The `@mixin` on `Proxy` should take care of that.

Reply:
I dont understand the `@mixin`, could you tell me how that works?

Reply:
From the release notes of https://github.com/vimeo/psalm/releases/tag/3.8.2

> Support for the @mixin docblock annotation, which allows a class to proxy the methods and properties of another.
>
> I also made a slight change in how magic getters & setters are dealt with – Psalm will emit an UndefinedMagicProperty if you define a docblock @property string $foo and then request the unlisted property $bar. If you wanted this behaviour previously you had to also annotate your class with @psalm-seal-properties. This is now assumed if you provide a @mixin or @property annotation for the given class. In the event the property is missing, Psalm now emits an UndefinedMagicProperty issue that you can suppress in a more granular fashion.

Reply:
So again, you are correct that psalm will understand that. But the `&TModel` is for PHPStorm. 

I think this change is valid.

Reply:
As far as I can see, PHPstorm understands mixin? (not a phpstorm user myself though, so I can't test)

https://youtrack.jetbrains.com/issue/WI-1730
https://stackoverflow.com/a/48298989/1149495